### PR TITLE
MGMT-2337 Remove create, delete and monitor from job API

### PR DESCRIPTION
**Pull request by filanov**
Those functions are not used outside the package
Local and remote job implement the same interface so there is no reason to test twice with a mock that is doing the same thing

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -835,6 +835,7 @@ github.com/opencontainers/runtime-tools v0.0.0-20181011054405-1d69bd0f9c39/go.mo
 github.com/openshift-online/ocm-sdk-go v0.1.130 h1:2hnPXJSgkt0LSSNzO2lvJuFBq3rqZ01ntTMUua0D5fo=
 github.com/openshift-online/ocm-sdk-go v0.1.130/go.mod h1:3i3a/LEYtC7DMor3N94PTPn1+0qq+Fk+iLjt1Z0WVCs=
 github.com/openshift/api v0.0.0-20200205133042-34f0ec8dab87/go.mod h1:fT6U/JfG8uZzemTRwZA2kBDJP5nWz7v05UHnty/D+pk=
+github.com/openshift/assisted-installer v1.0.9 h1:hTAId3MU7/TMxK2v1neQ76JVghXtyOBkCaJMbkmy2TA=
```
Comment (danielerez) [inline go.sum:838]: Is it related to this PR?

**Comment by danielerez** [comment]
lgtm
Minor comment inline.

File: go.sum (hunk 1 above)
Comment (filanov) [reply go.sum:838]: Not really. i will remove it.

**Comment by filanov** [comment]
Had to fix a commit message because of a tab

**Pull request merged**